src/lib/utils/hooks/useGetInfoFromIds.ts

```diff
@@ -0,0 +1,125 @@
+  name: 'Unknown', //	The name of the entity
+  id: 'unknown', //	The id of the entity
+  type: 'unknown', //	The type of the entity
+  versionNumber: 0, //	The version number of the entity
+  versionLabel: 'placeholder', //	The user defined version label of the entity
+  benefactorId: 0, //	The ID of the entity that this Entity's ACL is inherited from.
+  createdOn: 'null', //	The date this entity was created.
+  modifiedOn: 'null', //	The date this entity was last modified.
+  createdBy: 'null', //	The ID of the user that created this entity.
+  modifiedBy: 'null', //	The ID of the user that last modified this entity.
+}
+
+const getEntityHeaderItems = async (
+  lookupList: ReferenceList,
+  token: string | undefined,
+): Promise<EntityHeader[]> => {
+  const newData = await getEntityHeader(lookupList as ReferenceList, token)
+  const notFound = lookupList.filter(
+    item =>
+      newData.results.map(item => item.id).indexOf(item.targetId) == -1,
+  )
+  const notFoundPlaceholders = notFound.map(item => ({
+    ...entityHeaderTemplate,
+    id: item.targetId,
+    name: `${item.targetId}`,
+  }))
+
+  return [...newData.results, ...notFoundPlaceholders]
+}
+
+const getUserProfileItems = async (
+  lookupList: string[],
+  token: string | undefined,
+): Promise<UserProfile[]> => {
+  const newData = await getUserProfileWithProfilePicAttached(
+    lookupList as string[],
+    token,
+  )
+  const notFound = lookupList.filter(
+    item => newData.list.map(item => item.ownerId).indexOf(item) == -1,
+  )
+  const notFoundPlaceholders = notFound.map(item => ({
+    ...UserProfileTemplate,
+    ownerId: item,
+    name: `Unknown User (${item})`,
+  }))
+
+  return [...newData.list, ...notFoundPlaceholders]
+}
+
+// React hook to get user profiles or entities
+//******************************************************************************************* */
+export default function useGetInfoFromIds<T extends EntityHeader | UserProfile>(
+  props: UseGetInfoFromIdsProps,
+) {
+  const { token, ids, type } = props
+  const [data, setData] = useState<Array<T>>([])
+
+  const idProp = (type: HookType) =>
+    type === 'USER_PROFILE' ? 'ownerId' : 'id'
+
+  useEffect(() => {
+
+    const getData = async () => {
+      // look at current list of data, see if incoming ids has new data,
+      // if so grab those ids
+      const curList = data.map(el => el[idProp(type)])
+      const incomingList = ids.filter(
+        el => el !== SynapseConstants.VALUE_NOT_SET,
+      )
+      const newValues = without(incomingList, ...curList)
+      if (newValues.length > 0) {
+        try {
+          const newIds = Array.from<string>(newValues)
+          const newReferences: LookupRequestType[] =
+            type === 'USER_PROFILE'
+              ? newIds
+              : newIds.map(el => ({ targetId: el }))
+          const newReferencesChunks = chunk(newReferences, 45)
+
+          for (const newReferences of newReferencesChunks) {
+            const newData =
```

Comment:
Just thinking out loud, if we have more types that this hook supports in the future then we'll want to make this a switch statement and possibly put these `get...Items` functions in another file, but for now this works, same with idProp from above